ocaml/typing/persistent_env.ml

```diff
@@ -20,15 +20,28 @@ open Misc
 open Cmi_format
 
 module CU = Compilation_unit
-module Consistbl = Consistbl.Make (CU.Name) (CU)
+module Impl = struct
+  (* The implementation compilation unit for some import, if known *)
+  type t =
+    | Unknown_argument (* The import is a parameter module *)
+    | Known of CU.t
+
+  let and_crc_of_import_info info =
```

Comment:
That gives me an even better idea: just chuck the `Import_info.Intf.t` directly into the `Consistbl`.

Reply:
Wait, no, that's not quite right. Essentially, something needs to play the role of an entry in the consistbl _minus the CRC_.

Reply:
I've made the code more idiomatic by wrapping the function in a module called `With_crc` and renaming the function from `and_crc_of_import_info` to simply `of_import_info` (so that its full name is `Impl.With_crc.of_import_info`). Whether or not we rename `Impl`, the `Consistbl` API requires that we have _something_ with _exactly_ the data in `Impl.t`. (In particular, when `-as-parameter-for` comes around, we _don't_ want to add the `arg_for` field, so this shouldn't be exactly `Cmi_format.kind`.)

Reply:
Ditched this code along with the rest of the `Impl` module.